flagscale/train/hetero/p2p_communication.py

```diff
@@ -41,13 +41,13 @@ def warm_up_comm_group_hetero(config: ModelParallelConfig):
     to_send_tensor= torch.empty(
             tensor_shape,
             requires_grad=True,
-            device=torch.cuda.current_device(),
+            device=torch.cuda.current_device() if pp_groups[0].name() != "gloo" else torch.device("cpu"),
             dtype=config.pipeline_dtype,
         )
     to_recv_tensor= torch.empty(
             tensor_shape,
             requires_grad=True,
-            device=torch.cuda.current_device(),
+            device=torch.device("cpu") if pp_groups[0].name() != "gloo" else torch.cuda.current_device(),
```

Comment:
Why is this line different from line 44?

Reply:
Here is a fault. Change this line to the same code as line 44.